Dockerfiles/agent/entrypoint/10-dogstatsd-socket.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Delete the dogstatsd unix socket if present
+# FIXME: move that logic to dsd itself
+
+if [[ -z "${DD_DOGSTATSD_SOCKET}" ]]; then
+    export DD_DOGSTATSD_NON_LOCAL_TRAFFIC=1
+elif [[ -e "${DD_DOGSTATSD_SOCKET}" ]]; then
+    if [[ -S "${DD_DOGSTATSD_SOCKET}" ]]; then
```

Comment:
When asked to exit s6 sends `SIGTERM` to all processes (then `SIGKILL` after a moment) and reaps zombies. So we should not get any zombie dsds. But if the agent is recieves a `SIGKILL`, it will exit without cleaning up the socket. That makes a socket inode with nothing listening on it, which will prevent dsd from starting again :/